pkg/api/export/exporter.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is largely derived from the upstream open shift command:
+//  https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/exporter.go
+
+package export
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/registry/controller"
+	"k8s.io/kubernetes/pkg/registry/endpoint"
+	"k8s.io/kubernetes/pkg/registry/namespace"
+	"k8s.io/kubernetes/pkg/registry/node"
+	"k8s.io/kubernetes/pkg/registry/persistentvolume"
+	"k8s.io/kubernetes/pkg/registry/persistentvolumeclaim"
+	"k8s.io/kubernetes/pkg/registry/pod"
+	"k8s.io/kubernetes/pkg/registry/resourcequota"
+	"k8s.io/kubernetes/pkg/registry/secret"
+	"k8s.io/kubernetes/pkg/registry/serviceaccount"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+var ErrExportOmit = fmt.Errorf("object is omitted")
+
+// Exporter knows how to transform a runtime.Object into one that has had cluster specific fields stripped.
+type Exporter interface {
+	AddExportOptions(*pflag.FlagSet)
+	Export(obj runtime.Object, exact bool) error
+}
+
+type DefaultExporter struct{}
+
+func (e *DefaultExporter) AddExportOptions(flags *pflag.FlagSet) {
+}
+
+func exportObjectMeta(objMeta *api.ObjectMeta, exact bool) {
+	objMeta.UID = ""
+	if !exact {
+		objMeta.Namespace = ""
+	}
+	objMeta.CreationTimestamp = unversioned.Time{}
+	objMeta.DeletionTimestamp = nil
+	objMeta.ResourceVersion = ""
+	objMeta.SelfLink = ""
+	if len(objMeta.GenerateName) > 0 && !exact {
+		objMeta.Name = ""
+	}
+}
+
+func (e *DefaultExporter) Export(obj runtime.Object, exact bool) error {
+	if meta, err := api.ObjectMetaFor(obj); err == nil {
+		exportObjectMeta(meta, exact)
+	} else {
+		glog.V(4).Infof("Object of type %v does not have ObjectMeta: %v", reflect.TypeOf(obj), err)
+	}
+	switch t := obj.(type) {
```

Comment:
I would strongly prefer to see something added to the strategy (like the list, get, create, etc behaviors). You can even add a default to the generic one and get most of this for free, it looks like.